### PR TITLE
Robin/rename bugsnag notifier class to bugsnag client

### PR DESCRIPTION
**Pull request by robinmacharg**
## Goal

Rename the `BugsnagNotifier` class to `BugsnagClient`.

## Design

XCode refactor of the principle class, search/refactor instances of `notifier` to `client` appropriately.

## Changeset

Various files:

```
M	Source/Bugsnag.m
R098	Source/BugsnagNotifier.h	Source/BugsnagClient.h
R099	Source/BugsnagNotifier.m	Source/BugsnagClient.m
M	Source/BugsnagConfiguration.h
M	Source/BugsnagConfiguration.m
M	Source/BugsnagErrorReportApiClient.m
M	Source/BugsnagKSCrashSysInfoParser.m
M	Source/BugsnagSessionTrackingPayload.m
M	Source/BugsnagSink.m
M	Tests/BugsnagBaseUnitTest.m
```

## Tests

Code compiles, Unit Tests all pass.

### Outstanding Questions

The library is still conceptually a 'Notifier';  The class has been renamed to better indicate its purpose (as a client communicating with the Bugsnag backend).  Are the changes appropriate, have any instances of 'notifier' in the source __not__ been renamed when they should have been?

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [X] Final review

- [ ] The correct target branch has been selected (`master` for fixes, `next` for
  features)
- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [X] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: Tests/BugsnagBaseUnitTest.m [hunk 1]
```diff
@@ -26,7 +26,7 @@ @implementation BugsnagBaseUnitTest
  *
  * We take the former approach.
  *
- * @param willNotify Whether the notifier should actually send the event to the server
+ * @param willNotify Whether the Notifier should actually send the event to the server
```
Comment (tomlongridge) [inline Tests/BugsnagBaseUnitTest.m]:
```suggestion
 * @param willNotify Whether the notifier should actually send the event to the server
```

File: Source/BugsnagKSCrashSysInfoParser.m [hunk 2]
```diff
@@ -103,19 +103,19 @@
     
     BSGDictSetSafeObject(app, codeBundleId, @"codeBundleId");
     
-    NSString *notifierType;
+    NSString *clientType;
```
Comment (tomlongridge) [inline Source/BugsnagKSCrashSysInfoParser.m]: I know it's just a variable name, but can you revert this too?
